ext/test/fail_fs/fail_fs.c

```diff
@@ -269,7 +269,7 @@ fail_file_read(WT_FILE_HANDLE *file_handle,
 			    fail_fh->iface.name,
 			    (uintmax_t)len, (uintmax_t)offset,
 			    wtext->strerror(wtext, NULL, errno));
-			ret = (nr == 0 ? WT_ERROR : errno);
+			ret = (nr == 0 ? WT_DATA_CORRUPTION : errno);
```

Comment:
I was hoping that we could restrict `WT_DATA_CORRUPTION` to only be returned by `wiredtiger_open` for now - could we achieve that by converting `WT_DATA_CORRUPTION` into WT_ERROR in the `API_END` macro?

That way applications won't need to change all of their WiredTiger API callers to be able to handle a new error return code.

Reply:
It cannot be added to `API_END` because not all callers send in a `ret` variable that can be assigned. There are `API_END_RET` and friends that take a `ret` variable (although it doesn't appear it must - we just use it that way now).